platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java

```diff
@@ -0,0 +1,214 @@
+            return new SwirldsPlatform(this);
+        } finally {
+
+            // Future work: eliminate the static variables that require this code to exist
+            if (blocks.firstPlatform()) {
+                MetricsDocUtils.writeMetricsDocumentToFile(
+                        getGlobalMetrics(),
+                        getPlatforms(),
+                        blocks.platformContext().getConfiguration());
+                getMetricsProvider().start();
+            }
+        }
+    }
+
+    /**
+     * Provide an event hasher in place of the platform's default event hasher.
+     *
+     * @param eventHasher the event hasher to use
+     * @return this builder
+     */
+    @NonNull
+    public PlatformComponentBuilder withEventHasher(@NonNull final EventHasher eventHasher) {
+        throwIfAlreadyUsed();
+        this.eventHasher = Objects.requireNonNull(eventHasher);
+        return this;
+    }
+
+    /**
+     * Build the event hasher if it has not yet been built. If one has been provided via
+     * {@link #withEventHasher(EventHasher)}, that hasher will be used. If this method is called more than once, only
+     * the first call will build the event hasher. Otherwise, the default hasher will be created and returned.
+     *
+     * @return the event hasher
+     */
+    @NonNull
+    public EventHasher buildEventHasher() {
+        if (eventHasher == null) {
+            eventHasher = new DefaultEventHasher(blocks.platformContext());
+        }
+        return eventHasher;
+    }
+
+    /**
+     * Provide an internal event validator in place of the platform's default internal event validator.
+     *
+     * @param internalEventValidator the internal event validator to use
+     * @return this builder
+     */
+    @NonNull
+    public PlatformComponentBuilder withInternalEventValidator(
+            @NonNull final InternalEventValidator internalEventValidator) {
+        throwIfAlreadyUsed();
+        this.internalEventValidator = Objects.requireNonNull(internalEventValidator);
+        return this;
+    }
+
+    /**
+     * Build the internal event validator if it has not yet been built. If one has been provided via
+     * {@link #withInternalEventValidator(InternalEventValidator)}, that validator will be used. If this method is called
+     * more than once, only the first call will build the internal event validator. Otherwise, the default validator
+     * will be created and returned.
+     *
+     * @return the internal event validator
+     */
+    @NonNull
+    public InternalEventValidator buildInternalEventValidator() {
+        if (internalEventValidator == null) {
+            final boolean singleNodeNetwork = blocks.initialState()
+                            .get()
+                            .getState()
+                            .getPlatformState()
+                            .getAddressBook()
+                            .getSize()
```

Comment:
`getAddressBook` is annotated `@Nullable`. I'm not sure under what conditions it would be `null`... does this need consideration to avoid NPE calling `getSize` here?